include/hipCPU/hip/hip_runtime.h

```diff
@@ -783,7 +783,7 @@ hipError_t hipGetDeviceProperties(hipDeviceProp_t* p_prop, int device)
   std::string device_name = "hipCPU OpenMP host device";
   int max_dim = std::numeric_limits<int>::max();
 
-  strncpy(p_prop->name, device_name.c_str(), 256);
+  strncpy(p_prop->name, device_name.c_str(), sizeof(p_prop->name)-1);
```

Comment:
I might be a bit stupid here because I don't use those kind of functions often, but could you explain why we need the `-1`? I suspect this addresses the case where the string is longer than our buffer, and we want to leave the final byte in the buffer free to hold the `'\0'`. But I think `strncpy` won't add a null if the buffer is too small, so wouldn't we need to additionally add that null explicitly in that case?

Reply:
You're right, that would still miss the null terminator. I've made an update that avoids `strncpy` altogether.